### ISSUE TITLE
ColorButton adornment should have role="presentation" to pass accessibility scan

### ISSUE THREAD

**Issue by ryan-mcginty-alation** [viclafouch/mui-color-input#38]
**Describe the bug**
Mui color input does not meet accessibility requirements due to "no discernible text" error for ColorButton adornment. Current implementation of labeled-by only works when the popover is open which isn't useful.

**To Reproduce**
Run AXE accessibility tester via unit test or chrome plugin and note the error:

```
Expected the HTML found at $('.css-7fsjx7-MuiStack-root.MuiStack-root:nth-child(1) > .MuiFormControl-root.MuiTextField-root.MuiColorInput-TextField > .MuiInputBase-root.MuiOutlinedInput-root.MuiInputBase-colorPrimary > .MuiInputAdornment-root.MuiInputAdornment-positionStart.MuiInputAdornment-outlined > button') to have no violations:

<button class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-disableElevation MuiColorInput-Button  css-l6sfoi-MuiButtonBase-root-MuiButton-root" tabindex="0" type="button" style="background-color: rgb(0, 65, 107);">

Received:

"Buttons must have discernible text (button-name)"

Fix any of the following:
  Element does not have inner text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute
  Element's default semantics were not overridden with role="none" or role="presentation"
```

**Expected behavior**
In my opinion, since this is a purely visual interaction the color button could be set to have role="presentation" and the visually impaired user can simply enter the text value in the input to change the color.

**Screenshots**
<img width="1486" alt="Screenshot 2024-07-11 at 7 59 03 AM" src="https://github.com/viclafouch/mui-color-input/assets/82122037/660b00ed-7078-4b81-955c-1f2cec1037dc">

**Comment by viclafouch** [comment]
Hello ! 

I think it's better to pass an aria-label to the button, it's more convenient in this use case. Exemple "Open color palette.

**Comment by ryan-mcginty-alation** [comment]
That works for me, it seems the only way to do that is to completely re-wire the adornment right? Or am I missing something?

**Comment by viclafouch** [comment]
`AdornmentProps` added to the props of `MuiColorInput` should be the best options yes ;)

**Comment by Matt-Butler** [comment]
Have you considered updating the props to slotProps to include options for popper and adornment? This would align with the conventions in the latest MUI version.
